master/buildbot/status/web/base.py

```diff
@@ -403,6 +403,68 @@ def path_to_json_past_builds(request, builderName, number, filter_data=False):
     return url
 
 
+def get_query_branches_for_codebases(tags, codebases):
+    regex_branches = [
+        r'^(trunk)/',                 # Trunk
+        r'^(20[0-9][0-9].[0-9])\/',   # 2017.1/
+        r'^([0-9].[0-9])\/',          # 5.0/
+        r'^release\/([0-9].[0-9])/'   # release/4.6
+    ]
+
+    branches = map(lambda s: s.lower(), codebases.values())
+    codebase_keys = map(lambda s: s.lower(), codebases.keys())
+    query_branches = []
+
+    for branch, pattern in product(branches, regex_branches):
+        match = re.match(pattern, branch)
+        if match:
+            query_branches.append(match.group(1))
+
+    if not query_branches:
+        if 'unity' in codebase_keys:
+            query_branches = ['trunk']
+        else:
```

Comment:
Can we just skip this `else`? Then `query_branches` will be an empty list and only one `return` we will have in this function and the function will have one less branch and will be less complicated.

Reply:
We cannot. Without this `else` next `if not filter(...)` will be true and function return `['trunk']` instead `[]`
Without `else` tests will not pass.